### ISSUE TITLE
feature request: integration with `edgy.nvim`

### ISSUE THREAD

**Issue by olimorris** [stevearc/oil.nvim#122]
### Minimal.lua file to reproduce

```lua
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs({ "config", "data", "state", "cache" }) do
  vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  vim.fn.system({
    "git",
    "clone",
    "--filter=blob:none",
    "--single-branch",
    "https://github.com/folke/lazy.nvim.git",
    lazypath,
  })
end
vim.opt.runtimepath:prepend(lazypath)

-- install plugins
local plugins = {
  {
    "olimorris/onedarkpro.nvim",
    config = true,
  },
  {
    "stevearc/oil.nvim",
    config = true,
  },
  {
    "folke/edgy.nvim",
    very_lazy = true,
    opts = {
      left = {
        {
          ft = "oil",
          title = "File System",
        },
      },
    },
  },
}

require("lazy").setup(plugins, {
  root = root .. "/plugins",
})

vim.cmd([[colorscheme onedark]])

vim.keymap.set("n", "-", require("oil").open, { desc = "Open parent directory" })
vim.keymap.set("n", "_", require("oil").open_float, { desc = "Open parent directory" })
```

Run with `nvim --clean -u minimal.lua [some file]`

### Did you check existing requests?

- [X] I have searched the existing issues

### Describe the feature

With the launch of [edgy.nvim](https://github.com/folke/edgy.nvim) we have the ability to nicely manage windows layouts from within Neovim.

My immediate thought was trying it with `oil.nvim`. Unfortunately it doesn't quite work out of the box.

### Provide background

Running the `minimal.lua` config:

- Using `-` keymap, it works initially but pressing `q` to close, we get an error:
`E5108: Error executing lua: ...i/.dotfiles/.repro/plugins/edgy.nvim/lua/edgy/window.lua:81: Vim:E444: Cannot close last window`

- Using `_` keymap, we get the following error:
`E5108: Error executing lua: ...s/Oli/.dotfiles/.repro/plugins/oil.nvim/lua/oil/init.lua:339: BufWinEnter Autocommands for "*": Vim(append):Error executing lua callback: ...s/Oli/.dotfiles/.repro/plugins/oil.nvim/lua/oil/init.lua:326: title requires border to be set`

**Comment by directormac** [comment]
Just set the title it will work.
I got it working with lazyvim, the only bug i'm seeing is whenever i open a file it opens on the edgy window we need more api like open actions.select_prev where it should open on last active window, and close too.

**Comment by olimorris** [comment]
> Just set the title it will work.
> 
> I got it working with lazyvim, the only bug i'm seeing is whenever i open a file it opens on the edgy window we need more api like open actions.select_prev where it should open on last active window, and close too.

It threw the same error. Assuming you mean setting the title like `title = "File System"` in Edgy?

**Comment by olimorris** [comment]
> I got it working with lazyvim, the only bug i'm seeing is whenever i open a file it opens on the edgy window we need more api like open actions.select_prev where it should open on last active window, and close too.

@directormac yeah good suggestion and worthy of its own separate issue

**Comment by olimorris** [comment]
Well [try it out](https://github.com/olimorris/dotfiles) for yourself if you so wish...

**Comment by stevearc** [comment]
@olimorris I could use some more context on this. I haven't used edgy personally, but I gather it's a plugin for applying a specific window layout. What are you actually trying to do here? Because when I run the minimal example and press `-`, it looks like it just opens oil in the current window per normal. I get that pressing `q` throws an error, but I don't know what the expected behavior should be. Is it supposed to return to the previous buffer? I notice also that if I have two windows open and press `-`, it opens oil on the left in a new window, and `q` closes it. Is this also what it should be doing with a single window?

Basically: give me some more information about how this is supposed to work and what it should be doing, and also maybe provide some reasons why you think the fix needs to be in oil and not in edgy.

**Comment by olimorris** [comment]
> I get that pressing q throws an error, but I don't know what the expected behavior should be. Is it supposed to return to the previous buffer?

Edgy creates a window and `q` closes that window via: `vim.api.nvim_win_close(self.win, false)`.

Also worth noting that using `_` throws an error:

```
E5108: Error executing lua: ...s/Oli/.dotfiles/.repro/plugins/oil.nvim/lua/oil/init.lua:339: BufWinEnter Autocommands for "*": Vim(append):Error executing lua callback: ...s/Oli/.dotfiles/.repro/plugins/oil.nvim/lua/oil/init.lua:326: title requires border to be set
```

I'd expect this to open without an issue, in a separate window (which it does after you press enter on the error message). Then you can use `q` to close it as intended.

Regarding the former, I'm not sure that it is an Oil problem. I'd have to dive deeper into it. However the latter error...is there something that could be done to make the border optional?

**Comment by stevearc** [comment]
That error is fixed on the most recent version. LMK if you update and it's still present.

I'm confused about
> Edgy creates a window

I don't see any new window? When I press `-`, it opens oil in the current window like normal. Should it be opening a new window?

https://github.com/stevearc/oil.nvim/assets/506791/b7484d5d-d0f3-4fa6-9029-829fef612354

I feel like I still don't know what the expected behavior is here.

**Comment by olimorris** [comment]
If you run `:echo win_getid()` you should see a different window ID for the edgy and oil instance. I'm not au fait with how it works under the hood. I'm assuming there is an element of trickery as it detects the oil filetype and appears to create a separate window for oil to load in.

Ideally the output, with `_` should be:

<img width="1512" alt="Screen Shot 2023-06-18 at 10 28 22@2x" src="https://github.com/stevearc/oil.nvim/assets/9512444/c63ad190-e20b-4765-8802-90836061fae5">

But I'm still getting the error:

```
E5108: Error executing lua: ...s/Oli/.dotfiles/.repro/plugins/oil.nvim/lua/oil/init.lua:339: BufWinEnter Autocommands for "*": Vim(append):Error executing lua callback: ...s/Oli/.dotfiles/.repro/plugins/oil.nvim/lua/oil/init.lua:326: title requires border to be set
```

**Comment by sachnr** [comment]
you can set the borders to none now that removes the border error atleast
```lua
float = {
	border = "none",
},
```

**Comment by stevearc** [comment]
If your goal is to open Oil in a new window and use edgy to move it around into a side panel, then there's a pretty simple keymap for that:

```lua
vim.keymap.set("n", "-", "<cmd>vsplit | Oil<CR>", { desc = "Open parent directory in a split" })
```

I'm going to close this issue. If there are specific incompatibilities with edgy that could be addressed, feel free to open a new one.
